internal/venice-client-common/src/main/java/com/linkedin/venice/utils/AvroSupersetSchemaUtils.java

```diff
@@ -158,8 +159,18 @@ private static List<Schema.Field> mergeFieldSchemas(Schema existingSchema, Schem
       if (fieldInExistingSchema != null) {
         fieldBuilder.setSchema(generateSupersetSchema(fieldInExistingSchema.schema(), fieldInNewSchema.schema()))
             .setDoc(fieldInNewSchema.doc() != null ? fieldInNewSchema.doc() : fieldInExistingSchema.doc());
+        if (!fieldInNewSchema.hasDefaultValue() && fieldInExistingSchema.hasDefaultValue()) {
+          fieldBuilder.setDefault(getFieldDefault(fieldInExistingSchema));
+        }
       }
-      fields.add(fieldBuilder.build());
+      Schema.Field geneartedField = fieldBuilder.build();
+      LogManager.getLogger(AvroSupersetSchemaUtils.class)
+          .info(
+              "generated: {} {} {}",
```

Comment:
Can you update this log to have more information such as `generated: {fieldName}, new schema has default value: {}, ...`?

Reply:
Thanks for pointing out! This is actually a debug log, I removed it.